packages/graph/hash_graph/lib/graph/src/ontology/entity_type.rs

```diff
@@ -11,26 +11,222 @@ use crate::{
     store::query::{OntologyPath, ParameterType, QueryRecord, RecordPath},
 };
 
+/// A path to a [`EntityType`] field.
 #[derive(Debug, PartialEq, Eq)]
 pub enum EntityTypeQueryPath {
-    VersionId,
+    /// The [`BaseUri`] of a [`EntityType`].
+    ///
+    /// ```rust
+    /// # use serde::Deserialize;
+    /// # use serde_json::json;
+    /// # use graph::ontology::EntityTypeQueryPath;
+    /// let path = EntityTypeQueryPath::deserialize(json!(["baseUri"]))?;
+    /// assert_eq!(path, EntityTypeQueryPath::BaseUri);
+    /// # Ok::<(), serde_json::Error>(())
+    /// ```
+    ///
+    /// [`BaseUri`]: type_system::uri::BaseUri
+    BaseUri,
+    /// The version of the [`EntityType`].
+    ///
+    /// ```rust
+    /// # use serde::Deserialize;
+    /// # use serde_json::json;
+    /// # use graph::ontology::EntityTypeQueryPath;
+    /// let path = EntityTypeQueryPath::deserialize(json!(["version"]))?;
+    /// assert_eq!(path, EntityTypeQueryPath::Version);
+    /// # Ok::<(), serde_json::Error>(())
+    /// ```
+    ///
+    /// In addition to specifying the version directly, it's also possible to compare the version
+    /// with a `"latest"` parameter, which will only match the latest version of one
```

Comment:
Like for datatype, we could use `each` here 
```suggestion
    /// with a `"latest"` parameter, which will only match the latest version of each
```